pickle-processor/src/main/java/com/fourlastor/pickle/StatementConverter.kt

```diff
@@ -50,24 +50,39 @@ class StatementConverter(roundEnv: RoundEnvironment) {
     }
 }
 
+@Suppress("TYPEALIAS_EXPANSION_DEPRECATION")
 private fun RoundEnvironment.getStepDefinitions(): List<StepDefinition> {
-    val givens = getElementsAnnotatedWith(Given::class.java)
-            .filter { it.kind == ElementKind.METHOD }
-            .map { StepDefinition(it as ExecutableElement, Regex(it.getAnnotation(Given::class.java).value)) }
+    return ArrayList<StepDefinition>().apply {
```

Comment:
I think we can remove the initial empty list and do:
```kotlin
return getMethodsAnnotatedWith(...).map { ... } +
  getMethodsAnnotatedWith(...).map { ... } +
  ... +
  getMethodsAnnotatedWith(...).map { ... }
```

Also, maybe we can extract:

```kotlin
inline fun <reified T> getStepDefinitions() = getMethodsAnnotatedWith(T::class.java)
  .map { StepDefinition(it, Regex(it.getAnnotation(T::class.java).value) }
```

And have:
```kotlin
return getStepDefinitions<But>() +
  getStepDefinitions<Given>() +
  ... +
  getStepDefinitions<DeprecatedBut>()
```

Reply:
I wish we could do the second suggestion. I tried it couple of time and ended miserably 😊. I also wanted to make some of the take `vararg`

Unfortunately the annotation's value() function will not be available with generics. Since they don't share an interface, it is not possible. 

Will do the first suggestion

Reply:
Addressed here: https://github.com/fourlastor/pickle/pull/66/commits/0086716549a71404feb2d20003930650aea24cd1